tests/conftest.py

```diff
@@ -144,3 +148,101 @@ def _generate_calculation_node(process_state=ProcessState.FINISHED, exit_status=
         return node
 
     return _generate_calculation_node
+
+
+@pytest.fixture
+def create_empty_config_instance() -> Config:
+    """Create a temporary configuration instance.
+
+    This creates a temporary directory with a clean `.aiida` folder and basic configuration file. The currently loaded
+    configuration and profile are stored in memory and at are automatically restored at the end of this context manager.
+
+    :return: a new empty config instance.
+    """
+    import tempfile
+
+    from aiida.common.utils import Capturing
+    from aiida.manage import configuration
+    from aiida.manage.configuration import settings, load_profile, reset_profile
+
+    with tempfile.TemporaryDirectory() as dirpath:
```

Comment:
probably best to just use the `tmp_path` fixture here: https://docs.pytest.org/en/5.4.3/reference.html#std:fixture-tmp_path